File: AppTool/src/main/java/com/luoye/apptool/utils/LogUtils.java
package com.luoye.apptool.utils;

import android.util.Log;

/**
 * Created by: luoye
 * Time: 2022/8/17
 * user: 日志工具
 */
public class LogUtils {

    private static boolean isDebug = true;


    public static void setDebug(boolean debug) {
        isDebug = debug;
    }

    public static void i(String tag, String msg) {
        if (isDebug) Log.i(tag, msg);
    }

    public static void v(String tag, String msg) {
        if (isDebug) Log.v(tag, msg);
    }

    public static void e(String tag, String msg) {
        if (isDebug) Log.e(tag, msg);
    }

    public static void w(String tag, String msg) {
        if (isDebug) Log.w(tag, msg);
    }

    public static void d(String tag, String msg) {
        if (isDebug) Log.d(tag, msg);
    }
}

File: ktBase/src/main/java/com/luoye/ktbase/utils/LogUtils.kt
package com.luoye.ktbase.utils
import android.util.Log
import com.luoye.ktbase.BuildConfig


/**
 *
 * Created by: 
 * Time: 2022/11/30
 * user: 日志工具
 *
 */
open class LogUtils {
    
    companion object{
        private var isDebug = BuildConfig.DEBUG

        private var TAG="---LogUtils"

        fun setTAG(tag:String){ TAG=tag }

        fun setDebug(debug: Boolean) { isDebug = debug }

        fun i(tag: String, msg: String) { if (isDebug) Log.i(tag, msg) }

        fun v(tag: String, msg: String) { if (isDebug) Log.v(tag, msg) }

        fun e(tag: String, msg: String) { if (isDebug) Log.e(tag, msg) }

        fun w(tag: String, msg: String) { if (isDebug) Log.w(tag, msg) }

        fun d(tag: String, msg: String) { if (isDebug) Log.d(tag, msg) }

        fun i(msg: String) {
            if (isDebug) Log.i(TAG, msg)
        }

        fun v(msg: String) {
            if (isDebug) Log.v(TAG, msg)
        }

        fun e(msg: String) {
            if (isDebug) Log.e(TAG, msg)
        }

        fun w(msg: String) {
            if (isDebug) Log.w(TAG, msg)
        }

        fun d(msg: String) {
            if (isDebug) Log.d(TAG, msg)
        }
    }
}